mmeval/metrics/instance_seg.py

```diff
@@ -0,0 +1,288 @@
+            pred_instance_labels,
+            pred_instance_scores,
+            valid_class_ids=self.valid_class_ids,
+            class_labels=self.classes)
+
+        return ret_dict
+
+    def aggregate_predictions(self, masks, labels, scores, valid_class_ids):
+        """Maps predictions to ScanNet evaluator format.
+
+        Args:
+            masks (list[np.ndarray]): Per scene predicted instance masks.
+            labels (list[np.ndarray]): Per scene predicted instance labels.
+            scores (list[np.ndarray]): Per scene predicted instance scores.
+            valid_class_ids (tuple[int]): Ids of valid categories.
+
+        Returns:
+            list[dict]: Per scene aggregated predictions.
+        """
+        infos = []
+        for id, (mask, label, score) in enumerate(zip(masks, labels, scores)):
+            info = dict()
+            n_instances = mask.max() + 1
+            for i in range(n_instances):
+                file_name = f'{id}_{i}'
+                info[file_name] = dict()
+                info[file_name]['mask'] = (mask == i).astype(int)
+                info[file_name]['label_id'] = valid_class_ids[label[i]]
+                info[file_name]['conf'] = score[i]
+            infos.append(info)
+        return infos
+
+    def rename_gt(self, gt_semantic_masks, gt_instance_masks, valid_class_ids):
+        """Maps gt instance and semantic masks to instance masks for ScanNet
+        evaluator.
+
+        Args:
+            gt_semantic_masks (list[np.ndarray]): Per scene gt semantic masks.
+            gt_instance_masks (list[np.ndarray]): Per scene gt instance masks.
+            valid_class_ids (tuple[int]): Ids of valid categories.
+
+        Returns:
+            list[np.array]: Per scene instance masks.
+        """
+        renamed_instance_masks = []
+        for semantic_mask, instance_mask in zip(gt_semantic_masks,
+                                                gt_instance_masks):
+            unique = np.unique(instance_mask)
+            assert len(unique) < 1000
+            for i in unique:
+                semantic_instance = semantic_mask[instance_mask == i]
+                semantic_unique = np.unique(semantic_instance)
+                assert len(semantic_unique) == 1
+                if semantic_unique[0] < len(valid_class_ids):
+                    instance_mask[
+                        instance_mask ==
+                        i] = 1000 * valid_class_ids[semantic_unique[0]] + i
+            renamed_instance_masks.append(instance_mask)
+        return renamed_instance_masks
+
+    def instance_seg_eval(self,
+                          gt_semantic_masks,
+                          gt_instance_masks,
+                          pred_instance_masks,
+                          pred_instance_labels,
+                          pred_instance_scores,
+                          valid_class_ids,
+                          class_labels,
+                          options=None):
```

Comment:
It is better to merge the implementation in this function into `compute_metric`, which avoid passing a lot of parameters~

Reply:
fixed